### PR TITLE
Loading status service

### PR DESCRIPTION
**Pull request by tdesvenain**
fixes: when several http requests are sent in parallel, loading status is true once the first one is done

### CODE REVIEW

File: docs/reference/services.rst [hunk 1]
```diff
@@ -158,3 +158,41 @@ The cache can't store more than as many entries as set on `CACHE_MAX_SIZE` prope
 A `hits` property contains the hits statistics (number of hits by path).
 
 Cache service is massively used by `resource` and `comments` service. All get requests are cached and all create/update/delete requests revokes cache.
+
+
+Loading service
+---------------
+
+Loading service stores ids for what is currently loading. You declare here which loadings has started and finished.
```
Comment (Gagaro) [inline docs/reference/services.rst]: "which loadings *have*"

File: docs/reference/services.rst [hunk 2]
```diff
@@ -158,3 +158,41 @@ The cache can't store more than as many entries as set on `CACHE_MAX_SIZE` prope
...
+
+The service provides observables that emits when loading status changes. This is useful when you want to display a reactive loader.
+
+You give an id each 'thing' you mark as loaded using `start` method. You mark loading as finished using `finish` method.
```
Comment (Gagaro) [inline docs/reference/services.rst]: *to* each
using *the* start method
*the* finish method

File: docs/reference/services.rst [hunk 3]
```diff
@@ -158,3 +158,41 @@ The cache can't store more than as many entries as set on `CACHE_MAX_SIZE` prope
...
+
+`status` behavior subject changes when there is nothing loading left or if there is at least one thing loading.
```
Comment (Gagaro) [inline docs/reference/services.rst]: nothing loading left => nothing left to load ?

File: docs/reference/services.rst (hunk 2 above)
Comment (Gagaro) [reply docs/reference/services.rst]: use `begin` instead of `start`.

File: src/api.service.ts [hunk 4]
```diff
@@ -93,7 +98,7 @@ export class APIService {
 
   private error(err: HttpErrorResponse) {
     const error: Error = JSON.parse(err.error);
-    this.status.next({ loading: false, error: error });
+    this.loading.finish();
```
Comment (Gagaro) [inline src/api.service.ts]: Find a way to only finish the loading which failed?

File: src/authentication.service.ts [hunk 5]
```diff
@@ -122,4 +138,10 @@ export class AuthenticationService {
     }
     return headers;
   }
+
+  private error(err: HttpErrorResponse) {
+    const error: Error = JSON.parse(err.error);
+    this.loading.finish();
```
Comment (Gagaro) [inline src/authentication.service.ts]: Same than API.

File: src/loading.service.ts [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    } else {
+      const ids = this.loadingIds.getValue()
+        .filter((loadingId: string) => loadingId !== id);
+      this.loadingIds.next(ids);
+    }
+  }
+
+  isLoading(id: string): Observable<boolean> {
+    return this.loadingIds.map((loadingIds: string[]) => {
+      for (let loadingId of loadingIds) {
```
Comment (Gagaro) [inline src/loading.service.ts]: `indexOf` :rofl:

**Comment by tdesvenain** [comment]
I use interceptor now, this removes a lot of code and resolves the issue with finishing on error @Gagaro

**Pull request merged**